CMakeLists.txt

```diff
@@ -10,23 +10,31 @@ project(libunifex LANGUAGES CXX
 
 list(APPEND CMAKE_MODULE_PATH "${CMAKE_CURRENT_SOURCE_DIR}/cmake")
 
+include(project-is-top-level)
 include(unifex_options)
 include(unifex_env)
 include(unifex_flags)
 
 add_subdirectory(source)
 
-if(BUILD_TESTING OR UNIFEX_BUILD_EXAMPLES)
-  enable_testing()
-
-  include(gtest)
-  include(CTest)
+if(NOT PROJECT_IS_TOP_LEVEL)
+  return()
+endif()
```

Comment:
This isn't quite what I was expecting. If I understand correctly, with this change, unifex will not build _at all_ unless it's the top level project. I don't want that. It's fine if examples and tests aren't built if it's not a top level project, but I still want to support building unifex in a subdirectory.

Reply:
The library target unifex is still consumable by using `add_subdirectory` from another top level project. Note, that the library target is defined in call to `add_subdirectory(sources/)` prior to the early return. I will add a comment and remove the early return in favor of a longer if statement such as in`if (PROJECT_IS_TOP_LEVEL AND (BUILD_TESTING OR UNIFEX_BUILD_EXAMPLES))` etc...